src/Confluent.Kafka/Producer.cs

```diff
@@ -194,6 +194,150 @@ public void HandleDeliveryReport(Message message)
                 SetResult(message);
             }
         }
+        
+        /// <summary>
+        ///     Detailed messaged
+        /// </summary>
+        public struct ProduceRecord
```

Comment:
I didn't want the user to have both ProduceRecord<K,V> and ProduceRecord to popup in intelissense, but no problem for me to move it